### PR TITLE
Create menu ui

### PR DESCRIPTION
**Pull request by khanh957**
+ Task : Create UI menu App
+ Redmine : https://edu-redmine.sun-asterisk.vn/issues/17561

image

### CODE REVIEW

File: app/src/main/res/values/styles.xml [hunk 1]
```diff
@@ -8,4 +8,8 @@
         <item name="colorAccent">@color/colorAccent</item>
     </style>
 
+    <style name="MainMenu">
+        <item name="android:layout_width">30dp</item>
```
Comment (anhmv-0866) [inline app/src/main/res/values/styles.xml]: hardcode

File: app/src/main/res/values/styles.xml [hunk 2]
```diff
@@ -8,4 +8,8 @@
...
+        <item name="android:layout_height">30dp</item>
```
Comment (anhmv-0866) [inline app/src/main/res/values/styles.xml]: hardcode

File: app/src/main/res/layout/activity_main.xml [hunk 3]
```diff
@@ -3,6 +3,78 @@
...
+        android:layout_height="70dp"
+        app:layout_constraintBottom_toBottomOf="parent"
+        tools:layout_editor_absoluteX="0dp"
+        />
+
+    <FrameLayout
+        android:id="@+id/frameMain"
+        android:layout_width="match_parent"
+        android:layout_height="0dp"
+        android:layout_marginTop="16dp"
```
Comment (anhmv-0866) [inline app/src/main/res/layout/activity_main.xml]: hardcode

File: app/src/main/res/layout/activity_main.xml [hunk 4]
```diff
@@ -3,6 +3,78 @@
...
+        android:textSize="30sp"
+        app:layout_constraintEnd_toStartOf="@+id/imageViewSearch"
+        app:layout_constraintStart_toEndOf="@+id/imageViewLibrary"
+        app:layout_constraintTop_toTopOf="parent"
+        />
+
+    <ImageView
+        android:id="@+id/imageViewSearch"
+        style="@style/MainMenu"
+        android:layout_marginEnd="20dp"
```
Comment (anhmv-0866) [inline app/src/main/res/layout/activity_main.xml]: hardcode

File: app/src/main/res/layout/activity_main.xml [hunk 5]
```diff
@@ -3,6 +3,78 @@
...
+        app:layout_constraintBottom_toBottomOf="@+id/textViewMusic"
+        app:layout_constraintEnd_toEndOf="parent"
+        app:layout_constraintTop_toTopOf="@+id/textViewMusic"
+        app:layout_constraintVertical_bias="1.0"
+        />
+
+    <com.google.android.material.tabs.TabLayout
+        android:id="@+id/tabLayoutMusic"
+        app:tabMode="scrollable"
+        app:tabIndicatorColor="#8190A5"
```
Comment (anhmv-0866) [inline app/src/main/res/layout/activity_main.xml]: hardcode

File: app/src/main/java/com/sun_asterisk/soundcloud_71/screen/main/MainActivity.java [hunk 6]
```diff
@@ -1,14 +1,61 @@
...
+    private ViewPager viewPagerMusic;
 
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_main);
+
+        initView();
+
+        imageViewLibrary.setOnClickListener(this);
```
Comment (thaohq-0814) [inline app/src/main/java/com/sun_asterisk/soundcloud_71/screen/main/MainActivity.java]: tạo method initListeners() rồi cho vào

File: app/src/main/java/com/sun_asterisk/soundcloud_71/screen/main/MainActivity.java [hunk 7]
```diff
@@ -1,14 +1,61 @@
...
+        MainAdapter mainAdapter = new MainAdapter(getSupportFragmentManager());
+        viewPagerMusic.setAdapter(mainAdapter);
+        tabLayoutMusic.setupWithViewPager(viewPagerMusic);
+    }
+
+    @Override
+    public void onClick(View view) {
+        switch (view.getId()){
+            case R.id.imageViewLibrary:
+
```
Comment (thaohq-0814) [inline app/src/main/java/com/sun_asterisk/soundcloud_71/screen/main/MainActivity.java]: remove blank

File: app/src/main/java/com/sun_asterisk/soundcloud_71/screen/main/adapter/MainAdapter.java [hunk 8]
```diff
@@ -0,0 +1,38 @@
+package com.sun_asterisk.soundcloud_71.screen.main.adapter;
+
+import androidx.annotation.Nullable;
+import androidx.fragment.app.Fragment;
+import androidx.fragment.app.FragmentManager;
+import androidx.fragment.app.FragmentStatePagerAdapter;
+import java.util.ArrayList;
+import java.util.List;
+
+public class MainAdapter extends FragmentStatePagerAdapter {
```
Comment (thaohq-0814) [inline app/src/main/java/com/sun_asterisk/soundcloud_71/screen/main/adapter/MainAdapter.java]: sao k dùng FragmentPagerAdapter

File: app/src/main/res/layout/activity_main.xml [hunk 9]
```diff
@@ -3,6 +3,58 @@
...
+        app:layout_constraintVertical_bias="1.0"
+        />
+
+    <TextView
+        android:id="@+id/textViewMusic"
+        android:layout_width="wrap_content"
+        android:layout_height="wrap_content"
+        android:layout_marginTop="@dimen/dp_15"
+        android:text="@string/text_music"
+        android:textColor="#000000"
```
Comment (thaohq-0814) [inline app/src/main/res/layout/activity_main.xml]:
```suggestion
        android:textColor="@android:color/black"
```

File: app/src/main/res/layout/activity_main.xml [hunk 10]
```diff
@@ -3,6 +3,58 @@
...
+        />
+
+    <com.google.android.material.tabs.TabLayout
+        android:id="@+id/tabLayoutMusic"
+        app:tabMode="scrollable"
+        app:tabIndicatorColor="@color/colorIndicator"
+        app:tabSelectedTextColor="@color/colorTabSelected"
+        android:layout_width="match_parent"
+        android:layout_height="wrap_content"
+        android:layout_marginTop="16dp"
```
Comment (thaohq-0814) [inline app/src/main/res/layout/activity_main.xml]: hardcode

File: app/src/main/res/values/dimens.xml [hunk 11]
```diff
@@ -0,0 +1,8 @@
+<?xml version="1.0" encoding="utf-8"?>
+<resources>
+    <dimen name="dp_20">20dp</dimen>
+    <dimen name="dp_15">15dp</dimen>
+    <dimen name="sp_30">30sp</dimen>
+    <dimen name="dp_70">70dp</dimen>
+    <dimen name="dp_16">16dp</dimen>
+</resources>
```
Comment (thaohq-0814) [inline app/src/main/res/values/dimens.xml]: add new line

File: app/src/main/java/com/sun_asterisk/soundcloud_71/screen/main/MainActivity.java [hunk 12]
```diff
@@ -1,14 +1,61 @@
...
+        textViewMusic.setOnClickListener(this);
+        imageViewSearch.setOnClickListener(this);
+    }
+
+    @Override
+    public void onClick(View view) {
+        switch (view.getId()){
+            case R.id.imageViewLibrary:
+                break;
+
```
Comment (thaohq-0814) [inline app/src/main/java/com/sun_asterisk/soundcloud_71/screen/main/MainActivity.java]: remove blank

File: app/src/main/java/com/sun_asterisk/soundcloud_71/utils/Constant.java [hunk 13]
```diff
@@ -0,0 +1,7 @@
+package com.sun_asterisk.soundcloud_71.utils;
+
+public class Constant {
+    public static final String BASE_URL = "https://api.soundcloud.com/tracks?";
+    public static final String BASE_API_KEY = "&client_id=" + "Xv42GTmbVlNFS0kx6AC9EMPwCskcx7Vk";
```
Comment (thaohq-0814) [inline app/src/main/java/com/sun_asterisk/soundcloud_71/utils/Constant.java]: hide key
Ref: https://viblo.asia/p/cac-thu-thuat-de-giau-secret-keypassword-khoi-source-control-va-ung-dung-voi-ci-trong-lap-trinh-android-PwRGgmOnkEd#_cach-tot-hon-su-dung-gradleproperties-1

File: app/src/main/res/layout/activity_main.xml [hunk 14]
```diff
@@ -3,6 +3,57 @@
     xmlns:android="http://schemas.android.com/apk/res/android"
     android:layout_width="match_parent"
     android:layout_height="match_parent"
+    xmlns:app="http://schemas.android.com/apk/res-auto"
```
Comment (thaohq-0814) [inline app/src/main/res/layout/activity_main.xml]: format code

**Review by thaohq-0814 — APPROVED** [review]

File: app/src/main/java/com/sun_asterisk/soundcloud_71/screen/main/MainActivity.java [hunk 15]
```diff
@@ -1,14 +1,58 @@
...
+import androidx.viewpager.widget.ViewPager;
+import com.google.android.material.tabs.TabLayout;
 import com.sun_asterisk.soundcloud_71.R;
+import com.sun_asterisk.soundcloud_71.screen.main.adapter.MainAdapter;
 
-public class MainActivity extends AppCompatActivity {
+public class MainActivity extends AppCompatActivity implements View.OnClickListener {
+
+    private ImageView imageViewLibrary, imageViewSearch;
+    private TextView textViewMusic;
```
Comment (tuanlvt-1117) [inline app/src/main/java/com/sun_asterisk/soundcloud_71/screen/main/MainActivity.java]: check convention name

File: app/src/main/java/com/sun_asterisk/soundcloud_71/screen/main/MainActivity.java [hunk 16]
```diff
@@ -1,14 +1,58 @@
...
+    private TabLayout tabLayoutMusic;
```
Comment (tuanlvt-1117) [inline app/src/main/java/com/sun_asterisk/soundcloud_71/screen/main/MainActivity.java]: check convention name

File: app/src/main/java/com/sun_asterisk/soundcloud_71/screen/main/MainActivity.java [hunk 17]
```diff
@@ -1,14 +1,58 @@
...
+    private ViewPager viewPagerMusic;
```
Comment (tuanlvt-1117) [inline app/src/main/java/com/sun_asterisk/soundcloud_71/screen/main/MainActivity.java]: check convention name

File: app/src/main/java/com/sun_asterisk/soundcloud_71/screen/main/MainActivity.java [hunk 18]
```diff
@@ -1,14 +1,58 @@
...
 
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_main);
+
```
Comment (tuanlvt-1117) [inline app/src/main/java/com/sun_asterisk/soundcloud_71/screen/main/MainActivity.java]: remove blank

File: app/src/main/java/com/sun_asterisk/soundcloud_71/screen/main/adapter/MainAdapter.java [hunk 19]
```diff
@@ -0,0 +1,38 @@
...
+
+import androidx.annotation.Nullable;
+import androidx.fragment.app.Fragment;
+import androidx.fragment.app.FragmentManager;
+import androidx.fragment.app.FragmentPagerAdapter;
+import java.util.ArrayList;
+import java.util.List;
+
+public class MainAdapter extends FragmentPagerAdapter {
+    private List<String> titles = new ArrayList<>();
```
Comment (tuanlvt-1117) [inline app/src/main/java/com/sun_asterisk/soundcloud_71/screen/main/adapter/MainAdapter.java]: check convention name

File: app/src/main/java/com/sun_asterisk/soundcloud_71/screen/main/adapter/MainAdapter.java [hunk 20]
```diff
@@ -0,0 +1,38 @@
...
+    private List<Fragment> fragments = new ArrayList<>();
```
Comment (tuanlvt-1117) [inline app/src/main/java/com/sun_asterisk/soundcloud_71/screen/main/adapter/MainAdapter.java]: check convention name

File: app/src/main/java/com/sun_asterisk/soundcloud_71/screen/main/adapter/MainAdapter.java [hunk 21]
```diff
@@ -0,0 +1,38 @@
...
+    }
+
+    @Override
+    public Fragment getItem(int i) {
+        return fragments.get(i);
+    }
+
+    @Override
+    public int getCount() {
+        return titles.size();
```
Comment (tuanlvt-1117) [inline app/src/main/java/com/sun_asterisk/soundcloud_71/screen/main/adapter/MainAdapter.java]: check null

**Comment by tuanlvt-1117** [comment]
commented

**Review by tuanlvt-1117 — APPROVED** [review]

**Pull request merged**